modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -1958,8 +1958,17 @@ private void _addFileEntry(
 					(Map<String, Serializable>)values.get(
 						objectField.getI18nObjectFieldName());
 
-				localizedValues.put(
-					valueLanguageId, fileEntry.getFileEntryId());
+				for (Map.Entry<String, Serializable> entry :
+						localizedValues.entrySet()) {
+
+					if (GetterUtil.getLong(entry.getValue()) !=
+							dlFileEntry.getFileEntryId()) {
```

Comment:
Can you sort please? 

`if (dlFileEntry.getFileEntryId() != GetterUtil.getLong(entry.getValue())) {`

Reply:
Just done!